src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts

```diff
@@ -1,46 +1,362 @@
+      const filterElement = await webdriver.driver.$(
+        `.dropdown li[aria-label="${FilterTypeToLabelMap[filterTypeAsEnum]}"]`
+      );
+      expect(filterElement).not.toBeNull();
+      filterElement.click();
+      // wait 1000 ms for the action.
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
+  );
+};
+
+// ====================================
+
+const setBandFrequency = async (
+  webdriver: { driver: Driver | undefined },
+  bandIndex: number,
+  frequency: number
+) => {
+  const inputElement = await webdriver.driver
+    .$$('.band')
+    [bandIndex - 1].$$('label')[0]
+    .$('input');
+  await inputElement.setValue(frequency);
+  await inputElement.keys('Tab');
+  // wait 1000 ms for the action.
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+};
+
+export const givenBandFrequency = (
+  given: DefineStepFunction,
+  webdriver: { driver: Driver | undefined }
+) => {
+  given(
+    /^the frequency of band (\d+) is (\d+)Hz$/,
+    async (bandIndex: number, frequency: number) => {
+      await setBandFrequency(webdriver, bandIndex, frequency);
+    }
+  );
+};
+
+export const whenSetBandFrequency = (
+  when: DefineStepFunction,
+  webdriver: { driver: Driver | undefined }
+) => {
+  when(
+    /^I set the frequency of band (\d+) to (\d+)Hz$/,
+    async (bandIndex: number, frequency: number) => {
+      await setBandFrequency(webdriver, bandIndex, frequency);
+    }
+  );
+};
+
+export const whenSetBandFrequencyUsingArrows = (
+  when: DefineStepFunction,
+  webdriver: { driver: Driver | undefined }
+) => {
+  when(
+    /^I click on the (up|down) arrow of band (\d+) (\d+) times$/,
+    async (direction: string, bandIndex: number, times: number) => {
+      const button = await webdriver.driver
+        .$$('.band')
+        [bandIndex - 1].$(`.arrow-${direction}`);
+
+      for (let i = 0; i < times; i += 1) {
+        await button.click();
+        // wait 500 ms for the action. necessary
+        await new Promise((resolve) => setTimeout(resolve, 500));
+      }
+    }
+  );
+};
+
+// ====================================
+
+const setPreampGain = async (
+  webdriver: { driver: Driver | undefined },
+  position: string
+) => {
+  const element = await webdriver.driver.$(
+    '.sideBar input[name="Pre-Amplification Gain (dB)-range"]'
+  );
+  const coord = { x: 0, y: 0 };
+  if (position === 'top') {
+    coord.y = -100;
+  } else if (position === 'bottom') {
+    coord.y = 100;
+  }
+  element.dragAndDrop(coord);
+  // wait 1000 ms for the action.
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+};
+
+const setPreampGainNumber = async (
+  webdriver: { driver: Driver | undefined },
+  gain: number
+) => {
+  const inputElement = await webdriver.driver.$(
+    '.sideBar input[name="Pre-Amplification Gain (dB)-number"]'
+  );
+  await inputElement.setValue(gain);
+  await inputElement.keys('Tab');
```

Comment:
Would simulating a key press of `Enter` make more intuitive sense? I'm fine with leaving it as `Tab` as well

Reply:
It didn't work with Enter :')